packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1475,21 +1199,352 @@ class _DialState extends State<_Dial> with SingleTickerProviderStateMixin {
+                    validator: _validateHour,
+                    onSavedSubmitted: _handleHourSavedSubmitted,
+                    onChanged: _handleHourChanged,
+                  ),
+                  const SizedBox(height: 8.0),
+                  if (!hourHasError && !minuteHasError)
+                    ExcludeSemantics(
+                      // TODO(rami-a): localize 'Hour'
+                      child: Text('Hour', style: theme.textTheme.caption, maxLines: 1, overflow: TextOverflow.ellipsis),
+                    ),
+                ],
+              )),
+              Padding(
+                padding: const EdgeInsets.only(top: 20.0),
+                child: _StringFragment(timeOfDayFormat: timeOfDayFormat),
+              ),
+              Expanded(child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: <Widget>[
+                  const SizedBox(height: 8.0),
+                  _HourMinuteTextField(
+                    selectedTime: _selectedTime,
+                    isHour: false,
+                    style: hourMinuteStyle,
+                    validator: _validateMinute,
+                    onSavedSubmitted: _handleMinuteSavedSubmitted,
+                  ),
+                  const SizedBox(height: 8.0),
+                  if (!hourHasError && !minuteHasError)
+                    ExcludeSemantics(
+                      // TODO(rami-a): localize 'Minute'
+                      child: Text('Minute', style: theme.textTheme.caption, maxLines: 1, overflow: TextOverflow.ellipsis),
+                    ),
+                ],
+              )),
+              if (!use24HourDials) ...<Widget>[
+                const SizedBox(width: 12.0),
+                _DayPeriodControl(
+                  selectedTime: _selectedTime,
+                  orientation: Orientation.portrait,
+                  onChanged: _handleDayPeriodChanged,
+                ),
+              ]
+            ],
+          ),
+          if (hourHasError || minuteHasError)
+            Text(
+              // TODO(rami-a): localize 'Enter a valid time'
+              'Enter a valid time',
+              style: theme.textTheme.bodyText2.copyWith(color: theme.colorScheme.error),
+            )
+          else
+            const SizedBox(height: 2.0),
+        ],
+      ),
+    );
+  }
+}
+
+class _HourMinuteTextField extends StatefulWidget {
+  const _HourMinuteTextField({
+    Key key,
+    @required this.selectedTime,
+    @required this.isHour,
+    @required this.style,
+    @required this.validator,
+    @required this.onSavedSubmitted,
+    this.onChanged,
+  }) : super(key: key);
+
+  final TimeOfDay selectedTime;
+  final bool isHour;
+  final TextStyle style;
+  final FormFieldValidator<String> validator;
+  final ValueChanged<String> onSavedSubmitted;
+  final ValueChanged<String> onChanged;
+
+  @override
+  __HourMinuteTextFieldState createState() => __HourMinuteTextFieldState();
```

Comment:
Is it common to use __ instead of _ here?

Reply:
nope, this is an unfortunate side effect of the IDE, fixed